inference-engine/src/offline_transformations/src/pruning/shrink_weights.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+
+#include "pruning.hpp"
+#include "mask_attribute.hpp"
+
+#include <ngraph/pass/manager.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/log.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ShrinkWeights, "ShrinkWeights", 0);
+
+bool ngraph::pass::ShrinkWeights::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    int64_t reduced_weights_count{0};
+    int64_t total_weights_count{0};
+    for (const auto & node : f->get_ordered_ops()) {
+        // calculate shape for every node in graph as the input shape may change
+        // during Constant shrinking
+        node->validate_and_infer_types();
+
+        // TODO: constant can be shared across functions so we need to avoid consumers from other function
+        auto const_node = std::dynamic_pointer_cast<opset6::Constant>(node);
+        if (!const_node) continue;
+
+        const auto & const_shape = const_node->get_shape();
+        total_weights_count += shape_size(const_shape);
+
+        auto mask = getMask(const_node->output(0));
+        if (!mask) continue;
+
+        auto last_output = const_node->output(0);
+        auto consumers = last_output.get_target_inputs();
+
+        if (mask->is_shape_like()) {
+            // TODO: think about it
+            auto res = const_node->get_shape_val();
+            if (res.size() != mask->size()) {
+                throw ngraph_error("Ooops...");
```

Comment:
😄 I propose to have more extended message

Reply:
Done.